src/iexamine.cpp

```diff
@@ -2634,25 +2634,46 @@ void iexamine::harvest_plant( Character &you, const tripoint &examp, bool from_a
         }
     } else { // Generic seed, use the seed item data
         const itype &type = *seed->type;
-        here.i_clear( examp );
 
-        int skillLevel = round( you.get_skill_level( skill_survival ) );
         ///\EFFECT_SURVIVAL increases number of plants harvested from a seed
-        int plant_count = rng( skillLevel / 2, skillLevel );
-        plant_count *= here.furn( examp )->plant->harvest_multiplier;
-        plant_count = std::min( std::max( plant_count, 1 ), 12 );
-        const int seedCount = std::max( 1, rng( plant_count / 4, plant_count / 2 ) );
-        for( item &i : get_harvest_items( type, plant_count, seedCount, true ) ) {
-            if( from_activity ) {
-                i.set_var( "activity_var", you.name );
+        const int skill_level = round( you.get_skill_level( skill_survival ) );
+        std::list<item> results = harvest_items( type, here, examp, skill_level );
+        if( results.empty() ) {
+            item tmp = item( type.get_id(), calendar::turn );
+            add_msg( m_info, _( "You fail to harvest anything." ) );
+        } else {
+            for( item &i : results ) {
+                if( from_activity ) {
+                    i.set_var( "activity_var", you.name );
+                }
+                add_msg( m_info, _( "You harvest %1$s (%2$d)." ), i.type_name(), i.count() );
+                you.i_add_or_drop( i, 1 );
             }
-            here.add_item_or_charges( you.pos(), i );
         }
-        here.furn_set( examp, furn_str_id( here.furn( examp )->plant->transform ) );
+        ///\EFFECT_INT Intelligence caps survival skill gains from harvesting
+        const int max_harvest_skill = you.int_cur / 6 + 1;
+        ///\EFFECT_SURVIVAL decreases survival skill gain from harvesting (NEGATIVE)
+        const int max_exp = ( max_harvest_skill - you.get_skill_level( skill_survival ) );
+        // Award experience for harvesting attempt regardless of success
+        you.practice( skill_survival, rng( 1, max_exp ), max_harvest_skill );
         you.moves -= to_moves<int>( 10_seconds );
     }
 }
 
+std::list<item> iexamine::harvest_items( const itype &type, map &here, const tripoint &pos,
+        const int skill_level )
+{
+    std::list<item> items;
+    here.i_clear( pos );
+    const int plant_count = roll_remainder( std::max( rng_float( 4, skill_level ), 6.0 )
```

Comment:
This is a balance shift. 
```
// skill  0   1   2   3   4   5   6   7   8   9  10
// old   1.0 1.0 1.5 2.0 3.0 3.5 4.5 5.0 6.0 6.5 7.5
// new   2.0 2.5 3.0 3.5 4.0 4.5 5.0 5.3 5.5 5.6 5.7
```
Survival 10 producing 7.5x as much food as survival 1 doesn't make sense to me. Ditto every level of survival providing a linear increase in production.
This new formula increases the minimum production, decreases the maximum, and makes survival 7 produce 2.5x as much as survival 0, then a bit of diminishing returns up to survival 10.